src/actions/mod.rs

```diff
@@ -92,6 +98,10 @@ impl ActionHandler {
     pub fn init<O: Output>(&self, init_options: InitializationOptions, out: O) {
         trace!("init: {:?}", init_options);
 
+        if let Err(err) = self.infer_config_defaults(&mut *self.config.lock().unwrap()) {
```

Comment:
I've tried to catch every `Err` and surface it out so we won't get a panic there (updated now, got rid of the `.expect()` and `.unwrap()`). Should I still spawn a new thread for that here?